tests/test_downloader_middleware.py

```diff
@@ -132,3 +136,73 @@ def parse(self, response):
         )
         > 0
     )
+
+
+@ensureDeferred
+async def test_spm_conflict_smartproxy_setting(caplog):
+    settings = {
+        "ZYTE_API_TRANSPARENT_MODE": True,
+        "ZYTE_SMARTPROXY_APIKEY": "foo",
+        "ZYTE_SMARTPROXY_ENABLED": True,
+        **SETTINGS,
+    }
+    mws = dict(settings["DOWNLOADER_MIDDLEWARES"])
+    mws["scrapy_zyte_smartproxy.ZyteSmartProxyMiddleware"] = 610
+    settings["DOWNLOADER_MIDDLEWARES"] = mws
+    crawler = get_crawler(NamedSpider, settings_dict=settings)
+    await crawler.crawl()
+    assert crawler.stats.get_value("finish_reason") == "plugin_conflict"
+
+
+@ensureDeferred
+async def test_spm_conflict_smartproxy_spider_attr():
+    class SPMSpider(Spider):
+        name = "spm_spider"
+        zyte_smartproxy_enabled = True
```

Comment:
Do we have a test with a setting set to True, but the attribute set to False (I assume attr takes the precendence)?